core/src/trezor/ui/layouts/tt.py

```diff
@@ -0,0 +1,293 @@
+    text2.bold("Are you sure you want", "to skip the backup?")
+    text2.br_half()
+    text2.normal("You can back up your", "Trezor once, at any time.")
+
+    if (
+        await interact(
+            ctx,
+            Confirm(text1, cancel="Skip", confirm="Back up", major_confirm=True),
+            "confirm_backup",
+            ButtonRequestType.ResetDevice,
+        )
+        is CONFIRMED
+    ):
+        return True
+
+    confirmed = (
+        await interact(
+            ctx,
+            Confirm(text2, cancel="Skip", confirm="Back up", major_confirm=True),
+            "confirm_backup",
+            ButtonRequestType.ResetDevice,
+        )
+    ) is CONFIRMED
+    return confirmed
+
+
+def confirm_path_warning(ctx: wire.GenericContext, path: str) -> LayoutType:
+    text = Text("Confirm path", ui.ICON_WRONG, ui.RED)
+    text.normal("Path")
+    text.mono(*break_path_to_lines(path, 17))
+    text.normal("is unknown.", "Are you sure?")
+    return interact(
+        ctx,
+        Confirm(text),
+        "path_warning",
+        ButtonRequestType.UnknownDerivationPath,
+    )
+
+
+def _show_qr(
+    address: str,
+    desc: str,
+    cancel: str = "Address",
+) -> Confirm:
+    QR_X = const(120)
+    QR_Y = const(115)
+    QR_SIZE_THRESHOLD = const(63)
+    QR_COEF = const(4) if len(address) < QR_SIZE_THRESHOLD else const(3)
+    qr = Qr(address, QR_X, QR_Y, QR_COEF)
+    text = Text(desc, ui.ICON_RECEIVE, ui.GREEN)
+
+    return Confirm(Container(qr, text), cancel=cancel, cancel_style=ButtonDefault)
+
+
+def _split_address(address: str) -> Iterator[str]:
+    return chunks(address, 17)  # 18 on T1
+
+
+def _hex_lines(data: bytes) -> Iterator[str]:
+    hex_data = hexlify(data).decode()
+    if len(hex_data) >= 18 * 5:
+        hex_data = hex_data[: (18 * 5 - 3)] + "..."
+    return chunks(hex_data, 18)
+
+
+def _show_address(
+    address: str,
+    desc: str,
+    network: str = None,
+) -> Confirm:
+    text = Text(desc, ui.ICON_RECEIVE, ui.GREEN)
+    if network is not None:
+        text.normal("%s network" % network)
+    text.mono(*_split_address(address))
+
+    return Confirm(text, cancel="QR", cancel_style=ButtonDefault)
+
+
+def _show_xpub(xpub: str, desc: str, cancel: str) -> Paginated:
+    pages = []  # type: List[ui.Component]
+    for lines in chunks(list(chunks(xpub, 16)), 5):
+        text = Text(desc, ui.ICON_RECEIVE, ui.GREEN)
+        text.mono(*lines)
+        pages.append(text)
+
+    content = Paginated(pages)
+
+    content.pages[-1] = Confirm(
+        content.pages[-1],
+        cancel=cancel,
+        cancel_style=ButtonDefault,
+    )
+
+    return content
+
+
+async def show_address(
+    ctx: wire.GenericContext,
+    address: str,
+    address_qr: str = None,
+    desc: str = "Confirm address",
+    network: str = None,
+    multisig_index: int = None,
+    xpubs: Sequence[str] = [],
+) -> None:
+    is_multisig = len(xpubs) > 0
+    while True:
+        if (
+            await interact(
```

Comment:
Currently `show_address` does not send intermediate button requests. I think we intended to keep that behavior?

Reply:
Actually it does - in master `show_address` as well as `show_qr` send BR and the application code calls them from a while loop. So the current behavior is kept afaict.